src/goto-harness/memory_snapshot_harness_generator.h

```diff
@@ -69,6 +69,46 @@ class memory_snapshot_harness_generatort : public goto_harness_generatort
     override;
 
 protected:
+  /// User provided goto location: function name and (maybe) location number;
+  /// the structure wraps this option with a parser
+  struct entry_goto_locationt
+  {
+    irep_idt function_name;
+    optionalt<unsigned> location_number;
+
+    entry_goto_locationt(const std::string &cmdl_option);
+  };
+
+  /// User provided source location: file name and line number; the structure
+  /// wraps this option with a parser
+  struct entry_source_locationt
+  {
+    irep_idt file_name;
+    unsigned line_number;
+
+    entry_source_locationt(const std::string &cmdl_option);
+  };
+
+  /// Wraps the information needed to identify the entry point. Initializes via
+  /// either \ref entry_goto_locationt or \ref entry_source_locationt
+  struct entry_locationt
+  {
+    irep_idt function_name;
+    optionalt<goto_programt::const_targett> start_instruction;
+
+    void initialize_via_goto(
```

Comment:
I'd be more comfortable if these were builder/constructor functions, i.e. `entry_locationt make_from_goto_location(...);`, `entry_locationt make_from_source_location`

Reply:
Good idea: done.